### ISSUE TITLE
Parallel Simulator support

### ISSUE THREAD

**Issue by saikrishna321** [wix/detox#97]
as FBSimulator support parallel run, does detox also allow users to run tests in parallel sim?

**Comment by rotemmiz** [comment]
Het @saikrishna321 , not at the moment, but this is something we consider, that's one of the main reasons we chose fbsimctl.

**Comment by LeoNatan** [comment]
Xcode 9 / iOS 11 simulator runtime has native support for multiple concurrent simulators. Sounds to me like a good idea to see if we can remove fbsimctl use.

**Comment by fbartho** [comment]
2018 checkin! -- I'd very much hope for this feature!

**Comment by DanielMSchmidt** [comment]
I think it is currently work in progress: https://twitter.com/rotemmiz/status/938108324523986944

**Comment by fbartho** [comment]
@DanielMSchmidt -- I saw that thread back in December (and commented), but at the time, I didn't know who anybody was (I hadn't been following things). I lost track of the twitter thread, and came to GitHub to see if this feature was on the roadmap. I can't wait for @rotemmiz work to ship!! ❤️

**Comment by rotemmiz** [comment]
Hey guys, 

The development on parallel support have been delayed for other (internal) more urgent things (+ a week of marking the check box on the seasonal flu).

The main issue was Jest 21 at the time, which its lifecycle did not support initialization per process (as opposed to per suite). Jest 22 now offers parallelization with support of custom runner (jest-runner-mocha in our case). I will get back to the task soon (hopefully in the next couple of weeks)

**Comment by tomatobrown** [comment]
@rotemmiz are you saying that if we get parallel support working that we should be dependant on using Jest?

**Comment by rotemmiz** [comment]
Support has just landed in master with the merge of #609 .
This has no real relation to Jest, we just made Detox process safe. The critical part of choosing a free device to run on and then commiting it to just one process is the essence of the multi sim support. Jest supports parallelization with execution of multiple processes, but there are other test runners which do the same, like AVA. Unfortunately Mocha does not support parallelization.

**Comment by rotemmiz** [comment]
This will be shipped with Detox 7.4.0

**Comment by LeoNatan** [comment]
Please open a new issue for the missing documentation.

**Comment by rotemmiz** [comment]
It is part of the cli documentation
https://github.com/wix/detox/blob/master/docs/APIRef.DetoxCLI.md#test

--workers

You are right though, we need to add notes on what it actually does under the hood when running in this mode.
